core/src/main/java/org/jdbi/v3/core/EnumConfig.java

```diff
@@ -14,55 +14,36 @@
 package org.jdbi.v3.core;
 
 import java.lang.annotation.Annotation;
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
 import org.jdbi.v3.core.config.JdbiConfig;
 
 /**
  * Configuration for behavior related to {@link Enum}s.
  */
 public class EnumConfig implements JdbiConfig<EnumConfig> {
-    private static final Set<Class<? extends Annotation>> ALLOWED_HANDLINGS = new HashSet<>(Arrays.asList(EnumByName.class, EnumByOrdinal.class));
-    private Class<? extends Annotation> handling;
+    private Class<? extends Annotation> defaultQualifier;
 
     public EnumConfig() {
-        handling = EnumByName.class;
+        defaultQualifier = EnumByName.class;
     }
 
     private EnumConfig(EnumConfig other) {
-        handling = other.handling;
+        defaultQualifier = other.defaultQualifier;
     }
 
-    /**
-     * Applies to both binding and mapping.
-     *
-     * @return true if enums are handled by name, false if enums are handled by ordinal
-     */
-    @Deprecated
-    public boolean isEnumHandledByName() {
-        return handling == EnumByName.class;
+    public Class<? extends Annotation> getDefaultQualifier() {
```

Comment:
Using an annotation type feels odd to me. I get the sense @stevenschlansker that you're trying to design for extension here. However the set of options that plain JDK enums is strictly closed: name and ordinal.

If somebody wants to map/bind their enums in another way, we already have a registration procedure for that.

IMO this config option should just be a simple enum property, e.g. `EnumStrategy.BY_NAME` or `EnumStrategy.BY_VALUE`.